### PR TITLE
[RDY] Topic/onewire

### PR DESCRIPTION
**Pull request by Emantor**
Add onewire support to labgrid, needs a running onewire server at the moment.

WIP:
* Convert to managedResource
* Add to exporter
* Test over network

### CODE REVIEW

File: labgrid/protocol/OneWireProtocol.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import abc
+
+
+class OneWireProtocol(abc.ABC):
```
Comment (jluebbe) [inline labgrid/protocol/OneWireProtocol.py]: At the protocol level, this should be more generic. Maybe call it DigitalOutput instead?

File: labgrid/driver/onewiredriver.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import attr
+
+from ..factory import target_factory
+from ..resource import OneWirePort
+from .common import Driver
+
+@target_factory.reg_driver
+@attr.s
+class OneWireDriver(Driver):
```
Comment (jluebbe) [inline labgrid/driver/onewiredriver.py]: Corresponding to the change to the Protocol, this should be a OneWirePIODriver, implementing the DigitalOutput protocol.

File: labgrid/driver/powerdriver.py [hunk 3]
```diff
@@ -93,3 +94,26 @@ def fallback():
     def get(self):
         return self.backend.get(self.port.host, self.port.index)
 
+@target_factory.reg_driver
+@attr.s
+class OneWirePowerDriver(Driver, PowerProtocol):
```
Comment (jluebbe) [inline labgrid/driver/powerdriver.py]: This would be called DigitalOutputPowerDriver.

File: labgrid/resource/onewireport.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+from ..factory import target_factory
+from .common import Resource
+
+
+@target_factory.reg_resource
+@attr.s
+class OneWirePort(Resource):
+    portname = attr.ib(validator=attr.validators.instance_of(str))
+    url = attr.ib(validator=attr.validators.instance_of(str))
```
Comment (jluebbe) [inline labgrid/resource/onewireport.py]: This should be called OneWirePIO and have host and path properties. Also add a docstring giving an example path.

**Comment by Emantor** [comment]
This should be ready for review now, removing WIP.

File: labgrid/driver/powerdriver.py [hunk 5]
```diff
@@ -93,3 +94,28 @@ def fallback():
     def get(self):
         return self.backend.get(self.port.host, self.port.index)
 
+@target_factory.reg_driver
+@attr.s
+class DigitalOutputPowerDriver(Driver, PowerProtocol):
+    """NetworkPowerDriver - Driver using a networked power switch to control a target's power"""
```
Comment (jluebbe) [inline labgrid/driver/powerdriver.py]: Please update the docstring.

File: labgrid/driver/powerdriver.py [hunk 6]
```diff
@@ -93,3 +94,28 @@ def fallback():
...
+    bindings = {"onewire": OneWirePIODriver, }
```
Comment (jluebbe) [inline labgrid/driver/powerdriver.py]: Shouldn't this bind to a DigitalOutputProtocol? Maybe use "output" as the name.

File: labgrid/resource/onewireport.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import attr
+
+from ..factory import target_factory
+from .common import Resource
+
+
+@target_factory.reg_resource
+@attr.s
+class OneWirePIO(Resource):
```
Comment (jluebbe) [inline labgrid/resource/onewireport.py]: Add a docstring with example host/port values.

**Pull request merged**